portfolio/src/main/webapp/community.js

```diff
@@ -56,3 +56,20 @@ function createComment(comment, color) {
   commentDivElement.appendChild(document.createElement('br'));
   commentSection.appendChild(commentDivElement);
 }
+
+function deleteAllComments() {
+    let commentSection = document.getElementById('submitted-comments');
+    while (commentSection.lastChild) {
+      commentSection.removeChild(commentSection.lastChild);
+    }
+
+    fetch('/comments?commentRequestedNum='+Number.MAX_SAFE_INTEGER)
```

Comment:
It's probably not the most efficient solution, network traffic wise, to:
1) request that all comments be sent from the server to the browser, and..
2) send N individual deletion requests from the browser to the server.  (where N is the total number of comments)

That process results in N + 1 total requests, and it also involves the server sending the actual text content of all comments to the browser, which doesn't need it in this case.

There are a lot of potential ways to reduce the number of over-the-network service calls that need to be made:
- You could add another parameter to your delete-comment endpoint called something like "shouldDeleteAll" which is just a boolean, and if that is set, you ignore the "id" parameter and just delete all comments.  This way, you wouldn't need to fetch all comments in the first place.. so only one total call to the server =o
- You could change the "id" parameter to "ids" and let it be an array.. the con here is that you would still need to fetch all of the comments initially to get all of the ids.  But two total calls to the server isn't terrible.

There's another way.. possibly the most correct way, but it involves a deeper understanding of web API design, and I doubt that the tutorials mean for you to have to learn web practices to this depth, but I thought I'd share it as I only vaguely remembered it myself and think it's an interesting thing to know / could be useful later:
I think most web server APIs support a few more methods in addition to GET and POST.  The others are PUT (or PATCH) and DELETE.

The idea is that for something like comments, you would send a GET request to "/comments" when you want to retrieve one or more comments, you'd send a POST request to "/comments" when you want to create a new comment (like you're doing now), you'd send a PUT request to "/comments" when you want to update or edit a comment, and you'd send a DELETE request to "/comments" when you want to delete one or more comments.  It's kind of cool... notice how all of the requests are sent to the exact same URL path i.e. "/comments", but the method type dictates which action should be taken.

And Java servlets support this.  You can override the doDelete method to implement the DELETE action =o

Anyway, I want to emphasize that I'm not an expert on this stuff and that nothing I'm suggesting is definitively the "right" answer.  More than anything, I think the best thing to consider from what I've said is the very first part, which is that it would probably be best to save network traffic by not having to send N requests to the server when what you really want to do is telling it to delete "all" of the comments =)  However you would like to go about that is up to you!

Here's Google Cloud's official guidance on web API design, if you feel like reading more:
https://cloud.google.com/apis/design/standard_methods

Reply:
Added doDelete